gpkit/constraints/set.py

```diff
@@ -214,7 +215,9 @@ def reset_varkeys(self):
     def as_posyslt1(self, substitutions=None):
         "Returns list of posynomials which must be kept <= 1"
         posylist, self.posymap = [], []
-        for constraint in self:
+        for i, constraint in enumerate(self):
+            if not hasattr(constraint, "as_posyslt1"):
+                raise_badelement(self, i, constraint)
             posys = constraint.as_posyslt1(substitutions)
```

Comment:
feel free to reject, I see that maybe `constraint.as_posyslt1` could exist and happen to raise its own `AttributeError`, in which case catching is not appropriate.